src/meteodatalab/operators/regrid.py

```diff
@@ -404,3 +407,77 @@ def icon2rotlatlon(field: xr.DataArray) -> xr.DataArray:
     )
 
     return _icon2regular(field, dst, indices, weights)
+
+
+def _linear_weights(xy, uv):
+    """Compute indices and weights for barycentric linear interpolation."""
+    tri = Delaunay(xy)
+    simplex = tri.find_simplex(uv)
+    isfound = simplex != -1
+    vertices = np.take(tri.simplices, simplex, axis=0)
+    indices = np.where(isfound[:, None], vertices, 0)
+
+    # note that zero is a valid index
+    # however, a full line of zeros is interpreted as out of bounds
+    # in the interpolation step
+
+    temp = np.take(tri.transform, simplex, axis=0)
+    delta = uv - temp[:, 2]
+    bary = np.einsum("njk,nk->nj", temp[:, :2, :], delta)
+    wgts = np.hstack((bary, 1 - bary.sum(axis=1, keepdims=True)))
+    weights = np.where(isfound[:, None], wgts, 0)
+
+    return indices, weights
+
+
+def _cropped_domain(xy, uv, buffer=1000):
```

Comment:
I think it would make sense to provide types and a bit more doc on these functions. 
Even if declare private, for maintainability and increase its readability by developers. 
Arguments are passed to other internal functions and complexity is not so trivial anymore.

Reply:
Also the function is not just cropping the domain, since it is also computing the weights?

Reply:
Yes, it's essentially cropping the input domain before passing it to the computation of the weights to avoid creating triangles where no interpolation is done.